### PR TITLE
Split tests into groups for parallel execution in CI/CD runners

### PR DESCRIPTION
**Pull request by billschereriii**
Split tests into groups for parallel execution in CI/CD runners
Remove some unused import statements

### CODE REVIEW

**Review by al-rigazzi — COMMENTED** [review]
I like the divide-and-conquer approach. Nevertheless, the probability of hitting the pre-existing issue with Codecov is multiplied by a factor of ~3, thus I'd suggest we decide whether we are going to mark as passed the action even if codecov failed and, if that's the case, we do it as part of this PR.

**Comment by billschereriii** [comment]
Looks like all the tests ran in between 9 to 18 minutes, btw

**Comment by billschereriii** [comment]
Note: subdirectories `full_wlm` and `on_wlm` are intentionally skipped for CI/CD as we aren't running on HPC systems for these tests

**Comment by ankona** [comment]
Per @al-rigazzi comment about codecov... Do we want to do a minor rearrange of the build to avoid 3x the uploads? Perhaps we could:

1. run independent test step
2. store each codecov report as an artifact in a test step
3. split code coverage upload to run into step successor of (1)
4. [retrieve](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#downloading-or-deleting-artifacts) all artifacts and upload only one time

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -44,11 +44,12 @@ env:
 
 jobs:
   run_tests:
-    name: Run tests with ${{ matrix.os }}, Python ${{ matrix.py_v}}, RedisAI ${{ matrix.rai }}
+    name: Run tests ${{ matrix.subset }} with ${{ matrix.os }}, Python ${{ matrix.py_v}}, RedisAI ${{ matrix.rai }}
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
+        subset: [backends, slow_tests, group_a, group_b]
```
Comment (ankona) [inline .github/workflows/run_tests.yml:52]: Other than the opaque group names, this is simple & effective. LGTM as the first step toward test domination!

**Review by ankona — APPROVED** [review]
LGTM. Approving & assuming you'll want to split out anything about rearranging to upload codecov results fewer times into a new ticket.

**Pull request merged**